### PR TITLE
ci: build and push `ghcr.io/coder/coder-preview:main` 

### PR DESCRIPTION
**Pull request by matifali**
Build and push a docker image with the tag `ghcr.io/coder/coder-preview:main` on each commit merged to the `main`.
We average merging 10-20 PRs daily, so I am avoiding creating an `commit-sha` tag.

### CODE REVIEW

**Comment by bpmct** [comment]
Where is the edge tag specified? This looks like it pushes to latest.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -695,3 +695,47 @@ jobs:
...
+          set -euxo pipefail
+          go mod download
+          make gen/mark-fresh
+          export DOCKER_IMAGE_NO_PREREQUISITES=true
+          version="$(./scripts/version.sh)"
+          export CODER_IMAGE_BUILD_BASE_TAG="$(CODER_IMAGE_BASE=coder-base ./scripts/image_tag.sh --version "$version")"
+          make -j build/coder_linux_amd64
+          ./scripts/build_docker.sh \
+            --arch amd64 \
+            --target ghcr.io/coder/coder:latest \
```
Comment (matifali) [inline .github/workflows/ci.yaml:738]:
```suggestion
            --target ghcr.io/coder/coder:edge \
```

**Comment by matifali** [comment]
@bpmct It will be just one tag `edge` that will be replaced on each commit to `main`
Ok, I am open to moving it to `ghcr.io/coder/coder-preview:edge`

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -695,3 +695,47 @@ jobs:
           fi
 
           echo "Required checks have passed"
+
+  build:
```
Comment (deansheather) [inline .github/workflows/ci.yaml:699]:
```suggestion
  build-edge:
```

**Review by deansheather — APPROVED** [review]

**Comment by matifali** [comment]
@bpmct should I move it to `ghcr.io/coder/coder-preview:edge` or using `ghcr.io/coder/coder:edge` is fine?

**Comment by matifali** [comment]
OK. Changing it to `ghcr.io/coder/coder-preview:main` as it points to the `main` branch.

**Review by bpmct — APPROVED** [review]

**Pull request merged**